network/src/validator_network/network_builder.rs

```diff
@@ -81,16 +76,16 @@ pub enum AuthenticationMode {
     Mutual(x25519::PrivateKey),
 }
 
-/// Append the correct libranet protocols to the given base address, depending on
-/// the configured authentication mode.
-fn append_libranet_protocols(
-    addr: NetworkAddress,
-    auth_mode: &AuthenticationMode,
-) -> NetworkAddress {
-    match auth_mode {
-        AuthenticationMode::Unauthenticated => addr.append_test_protos(HANDSHAKE_VERSION),
-        AuthenticationMode::ServerOnly(key) | AuthenticationMode::Mutual(key) => {
-            addr.append_prod_protos(key.public_key(), HANDSHAKE_VERSION)
+impl AuthenticationMode {
+    /// Convenience method to retrieve the public key for the auth mode's inner
+    /// network identity key.
+    ///
+    /// Note: this only works because all auth modes are Noise-based.
+    pub fn public_key(&self) -> x25519::PublicKey {
+        match self {
+            AuthenticationMode::ServerOnly(key) | AuthenticationMode::Mutual(key) => {
+                key.public_key()
+            }
```

Comment:
this match feels hacky, but I guess there's no other way to write this in Rust?
Probably AuthenticationMode should be re-written as a struct: 

```rust
pub struct AuthenticationMode {
  mutual_authentication: bool,
  private_key: x25519::PrivateKey,
}
```